WebApi/WebApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using WebApi.Models.DTO;
+using WebApi.Models.DTO.PostDTOs;
+using WebApi.Services.Services_Interfaces;
+
+
+namespace WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : ControllerBase
+    {
+        private readonly IPostService _postService;
+        private readonly ILogger<PostController> _logger;
+        public PostController(ILogger<PostController> logger , IPostService postService)
+        {
+            _logger = logger;
+            _postService = postService;
+        }
+
+
+        [HttpGet]
+        public ActionResult<IQueryable<PostDTO>> GetAll([Required][FromHeader] int userID)
+        {
+            var result = _postService.GetAll();
+            if (result == null) return NotFound();
```

Comment:
należy wywalić sprawdzanie czy result jest null, bo zamiast null to zwracamy wyjatek
ma być samo return OK(result)
Tak w każdej akcji kontrolera